src/pages/Upsolve/upsolve.actions.js

```diff
@@ -0,0 +1,41 @@
+export async function codeforces(accessToken, vir, page) {
```

Comment:
Add the `upsolve.actions.js` page in the `/actions` folder in `/src` directory. Create a file name `Upsolve.js` and add all the API calls into this file.

Reply:
Done